milli/src/error.rs

```diff
@@ -256,11 +256,19 @@ only composed of alphanumeric characters (a-z A-Z 0-9), hyphens (-) and undersco
             Self::InvalidSortableAttribute { field, valid_fields } => {
                 let valid_names =
                     valid_fields.iter().map(AsRef::as_ref).collect::<Vec<_>>().join(", ");
-                write!(
-                    f,
-                    "Attribute `{}` is not sortable. Available sortable attributes are: `{}`.",
-                    field, valid_names
-                )
+
+                    match valid_names.is_empty() {
+                        true => write!(
```

Comment:
Instead of doing a `match boolean { true => , false => }` you should do a `if boolean { } else { }` 👍

Reply:
Can you explain to me a little, when should I opt to use pattern matching or conditionals?

Reply:
It is considered _bad style_ or _unidiomatic_ to match on booleans, maybe because if-else is _the_ keyword to use to match on this type, or maybe because it is not a good thing to directly write the `true` and `false` words.

Reply:
Cool, I agree @Kerollmops! 

Btw I was looking for a way to match for `SomeValueInBtreeSet` and `None`, `_`, but I didn't find something like that, that could work with `BTreeSet`.

Reply:
Oh I think you can do something like that;
```rust
if let Some(value) = set.get(key) {
    // do something
}

Reply:
I am not sure that I understand what you wanted to do, could you explain in more detail? Or maybe provide a pseudo code?

Reply:
```rust
match valid_names {
  HasValues => write!(...),
  EmptySet => write!(...),
}
```
Something like this (copied from meilisearch-rust):

```rust
   pub fn get_uid(&self) -> u64 {
        match self {
            Self::Enqueued { content } | Self::Processing { content } => *content.as_ref(),
            Self::Failed { content } => *content.as_ref(),
            Self::Succeeded { content } => *content.as_ref(),
        }
    }
```

Reply:
Ho ok, so something like this should have worked:
```rust
// You must match on the empty set first to make sure it is not caught by the unrestricted match.
match valid_names {
  [] => write!(...),
  values => write!(...),
}

// Another way to express this could be that way, but it is much more complex for no real gain.
match valid_names {
  [_, ..] @ values => write!(...),
  empty => write!(...),
}
```

You can read about [the @ _bindings_ in the official Rust book](https://doc.rust-lang.org/book/ch18-03-pattern-syntax.html?highlight=match#-bindings).

Reply:
Awesome, thanks for sharing that!